From 24d41d2bd3d98325b3800345f4ba27a334b3894b Mon Sep 17 00:00:00 2001
From: Himanshu Agarwal <himanaga@codeaurora.org>
Date: Wed, 4 Oct 2017 13:51:29 +0530
Subject: qcacld-3.0: Add sanity check for vdev id to prevent OOB access

Add sanity check for vdev id in wma_nlo_scan_cmp_evt_handler()
to prevent out of bound access of memory.

Change-Id: I0589765875fc03d2712f484b44c633face37bb90
CRs-Fixed: 2119403
---
 core/wma/src/wma_scan_roam.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/wma/src/wma_scan_roam.c b/core/wma/src/wma_scan_roam.c
index 62f7028..5454075 100644
--- a/core/wma/src/wma_scan_roam.c
+++ b/core/wma/src/wma_scan_roam.c
@@ -3851,6 +3851,10 @@ int wma_nlo_scan_cmp_evt_handler(void *handle, uint8_t *event,
 	WMA_LOGD("PNO scan completion event received for vdev %d",
 		 nlo_event->vdev_id);
 
+	if (nlo_event->vdev_id >= wma->max_bssid) {
+		WMA_LOGE("Invalid vdev id from firmware");
+		return -EINVAL;
+	}
 	node = &wma->interfaces[nlo_event->vdev_id];
 
 	/* Handle scan completion event only after NLO match event. */
-- 
cgit v1.1

